app/src/main/java/com/hedvig/app/service/push/managers/CrossSellNotificationManager.kt

```diff
@@ -0,0 +1,108 @@
+package com.hedvig.app.service.push.managers
+
+import android.app.PendingIntent
+import android.content.Context
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.app.TaskStackBuilder
+import com.google.firebase.messaging.RemoteMessage
+import com.hedvig.app.R
+import com.hedvig.app.feature.crossselling.ui.CrossSellData
+import com.hedvig.app.feature.crossselling.ui.detail.CrossSellDetailActivity
+import com.hedvig.app.feature.crossselling.usecase.GetCrossSellsUseCase
+import com.hedvig.app.service.push.DATA_MESSAGE_BODY
+import com.hedvig.app.service.push.DATA_MESSAGE_TITLE
+import com.hedvig.app.service.push.setupNotificationChannel
+import e
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import java.util.concurrent.atomic.AtomicInteger
+
+class CrossSellNotificationManager(
+    private val crossSellsUseCase: GetCrossSellsUseCase
+) {
+    fun sendCrossSellNotification(context: Context, remoteMessage: RemoteMessage) {
+        val title = remoteMessage.data[DATA_MESSAGE_TITLE]
+        val body = remoteMessage.data[DATA_MESSAGE_BODY]
+        val type = remoteMessage.data[CROSS_SELL_TYPE]
+
+        CoroutineScope(Dispatchers.IO).launch {
+            val crossSell = getCrossSell(type)
+            if (crossSell != null) {
+                createIntentAndNotify(
+                    context = context,
+                    title = title,
+                    body = body,
+                    crossSell = crossSell
+                )
+            } else {
+                e { "Could not find cross-sell with type: $type" }
```

Comment:
Should we fall back to showing a notification, but simply opening to the insurance-tab? 🤔

Reply:
Yes! 👍